clipper_admin/clipper_admin/deployers/tensorflow.py

```diff
@@ -156,16 +158,87 @@ def predict(sess, inputs):
     """
     # save predict function
     serialization_dir = save_python_function(name, func)
-    # save Tensorflow session
-    tf_sess_save_loc = os.path.join(serialization_dir, "tfmodel/model.ckpt")
-    try:
-        saver = tf.train.Saver()
-        save_path = saver.save(tf_sess, tf_sess_save_loc)
-    except Exception as e:
-        logger.warn("Error saving Tensorflow model: %s" % e)
-        raise e
-
-    logger.info("TensorFlow model saved at: %s " % save_path)
+    # save Tensorflow session or copy the saved model into the image
+    if isinstance(tf_sess_or_saved_model_path, Session):
+        tf_sess_save_loc = os.path.join(serialization_dir,
+                                        "tfmodel/model.ckpt")
+        try:
+            saver = tf.train.Saver()
+            save_path = saver.save(tf_sess_or_saved_model_path,
+                                   tf_sess_save_loc)
+        except Exception as e:
+            logger.warn("Error saving Tensorflow model: %s" % e)
+            raise e
+        logger.info("TensorFlow model saved at: %s " % save_path)
+    else:
+        # Check if its a frozen Graph or a saved tensorflow Model
+
+        # A typical Tensorflow model contains 4 files:
+        #  model-ckpt.meta: This contains the complete graph. [This contains a serialized MetaGraphDef protocol buffer.
+        #  model-ckpt.data-0000-of-00001: This contains all the values of variables(weights, biases, placeholders,gradients, hyper-parameters etc).
+        #  model-ckpt.index: metadata.
+        #  checkpoint: All checkpoint information
+
+        # Frozen Graph
+        # Single encapsulated file(.pb extension) without un-necessary meta-data, gradients and un-necessary training variables
+
+        if os.path.isdir(tf_sess_or_saved_model_path):
+            # Directory - Check for Frozen Graph or a Saved Tensorflow Model
+            is_frozen_graph = glob.glob(
+                os.path.join(tf_sess_or_saved_model_path, "*.pb"))
+            if (len(is_frozen_graph) > 0):
+                os.makedirs(os.path.join(serialization_dir, "tfmodel"))
+                try:
+                    shutil.copyfile(
```

Comment:
@kakru's suggestion

>I'd suggest using copytree instead of copyfile to keep the behavior consistent between model types (cf. line 207). Possible problem might arise if there is more than one file in the saved_model_path (check is for >0) and only the first .pb file is copied